connect/api/src/main/java/org/apache/kafka/connect/data/Struct.java

```diff
@@ -229,7 +229,11 @@ public void validate() {
             Object value = values[field.index()];
             if (value == null && (fieldSchema.isOptional() || fieldSchema.defaultValue() != null))
                 continue;
-            ConnectSchema.validateValue(fieldSchema, value);
+            try {
+                ConnectSchema.validateValue(fieldSchema, value);
+            } catch(DataException e) {
+                throw new DataException("Invalid value: null used for required field: "  + field.name() + ", schema type: " + fieldSchema.type());
```

Comment:
yes, I will keep the original exception message and add some additional field info on it.